include/rosdiscover-clang/Value/Int.h

```diff
@@ -0,0 +1,33 @@
+#pragma once
+
+#include <string>
+
+#include "Value.h"
+
+namespace rosdiscover {
+
+class IntegerLiteral : public virtual SymbolicString {
+public:
+  IntegerLiteral(int const &literal) : literal(literal) {}
+  ~IntegerLiteral() {}
+
+  static IntegerLiteral* create(int const &literal) {
+    return new IntegerLiteral(literal);
+  }
+
+  void print(llvm::raw_ostream &os) const override {
+    os << "\"" << literal << "\"";
+  }
+
+  nlohmann::json toJson() const override {
+    return {
+      {"kind", "integer-literal"},
+      {"literal", literal}
+    };
+  }
+
+private:
+  int const literal;
```

Comment:
This is where we _may_ want to use an APInt (arbitrary precision integer) rather than a relatively small fixed-size integer (to avoid problems with longs, etc.).